pkg/api/globalquery/server/server.go

```diff
@@ -16,11 +16,11 @@ type Server struct {
 }
 
 // New creates a new global-query API server
-func New(addr string, resolver hosts.Resolver, querier distributed.Querier, opts ...server.Option) *Server {
+func New(serviceName, addr string, resolver hosts.Resolver, querier distributed.Querier, opts ...server.Option) *Server {
```

Comment:
Maybe a naive question: If the service name is a const in the respective `conf` sub-package and always the same, why does it have to be a parameter for the instantiation?